gossip/infostore.go

```diff
@@ -146,26 +152,29 @@ func (is *infoStore) getInfo(key string) *info {
 // addInfo adds or updates an info in the infos map.
 //
 // Returns nil if info was added; error otherwise.
-func (is *infoStore) addInfo(key string, i *info) error {
+func (is *infoStore) addInfo(key string, i *Info) error {
 	// Only replace an existing info if new timestamp is greater, or if
 	// timestamps are equal, but new hops is smaller.
 	var contentsChanged bool
 	if existingInfo, ok := is.Infos[key]; ok {
-		if i.Timestamp < existingInfo.Timestamp ||
-			(i.Timestamp == existingInfo.Timestamp && i.Hops >= existingInfo.Hops) {
+		if iNanos, existingNanos := i.Value.Timestamp.WallTime, existingInfo.Value.Timestamp.WallTime; iNanos < existingNanos ||
```

Comment:
eww.  This is tough to read.  
`
if (i.Value.Timestamp.WallTime < existingInfo.Value.Timestamp.WallTime) || 
  (i.Value.Timestamp.WallTime == existingInfo.Value.Timestamp.WallTime && i.Hops >= existingInfo.Hops)
`